lib/ruby_lsp/listeners/definition.rb

```diff
@@ -74,8 +80,60 @@ def on_constant_read_node_enter(node)
         find_in_index(name)
       end
 
+      sig { params(node: Prism::InstanceVariableReadNode).void }
+      def on_instance_variable_read_node_enter(node)
+        handle_instance_variable_definition(node.name.to_s)
+      end
+
+      sig { params(node: Prism::InstanceVariableWriteNode).void }
+      def on_instance_variable_write_node_enter(node)
+        handle_instance_variable_definition(node.name.to_s)
+      end
+
+      sig { params(node: Prism::InstanceVariableAndWriteNode).void }
+      def on_instance_variable_and_write_node_enter(node)
+        handle_instance_variable_definition(node.name.to_s)
+      end
+
+      sig { params(node: Prism::InstanceVariableOperatorWriteNode).void }
+      def on_instance_variable_operator_write_node_enter(node)
+        handle_instance_variable_definition(node.name.to_s)
+      end
+
+      sig { params(node: Prism::InstanceVariableOrWriteNode).void }
+      def on_instance_variable_or_write_node_enter(node)
+        handle_instance_variable_definition(node.name.to_s)
+      end
+
+      sig { params(node: Prism::InstanceVariableTargetNode).void }
+      def on_instance_variable_target_node_enter(node)
+        handle_instance_variable_definition(node.name.to_s)
+      end
+
       private
 
+      sig { params(name: String).void }
+      def handle_instance_variable_definition(name)
+        entries = T.cast(@index[name], T.nilable(T::Array[RubyIndexer::Entry::InstanceVariable]))
```

Comment:
The more I see this pattern and the more I think we should make the method generic. See [this example](https://sorbet.run/#%23%20typed%3A%20strict%0A%0A%0Aclass%20Index%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20class%20Entry%3B%20end%0A%20%20class%20Member%20%3C%20Entry%3B%20end%0A%20%20class%20Method%20%3C%20Member%3B%20end%0A%20%20class%20InstanceVariable%20%3C%20Member%3B%20end%0A%0A%20%20sig%20%7B%20void%20%7D%0A%20%20def%20initialize%0A%20%20%20%20%40entries%20%3D%20T.let%28%7B%7D%2C%20T%3A%3AHash%5BString%2C%20T%3A%3AArray%5BEntry%5D%5D%29%0A%20%20end%0A%0A%20%20sig%20do%0A%20%20%20%20type_parameters%28%3AN%29%0A%20%20%20%20%20%20.params%28name%3A%20String%2C%20type%3A%20T%3A%3AClass%5BT.all%28T.type_parameter%28%3AN%29%2C%20Entry%29%5D%29%0A%20%20%20%20%20%20.returns%28T.nilable%28T.type_parameter%28%3AN%29%29%29%20%0A%20%20end%0A%20%20def%20entry%28name%2C%20type%29%0A%20%20%20%20entries%20%3D%20T.let%28%5B%5D%2C%20T%3A%3AArray%5BEntry%5D%29%0A%0A%20%20%20%20candidates%20%3D%20%40entries%5Bname%5D%0A%20%20%20%20return%20unless%20candidates%0A%0A%20%20%20%20candidates.each%20do%20%7Centry%7C%0A%20%20%20%20%20%20entries%20%3C%3C%20entry%20if%20entry.is_a%3F%28type%29%0A%20%20%20%20end%0A%0A%20%20%20%20T.unsafe%28entries%29%0A%20%20end%0Aend%0A%0Aindex%20%3D%20Index.new%0A%0Aentries%20%3D%20index.entry%28%22foo%22%2C%20Index%3A%3AEntry%29%0AT.reveal_type%28entries%29%0A%0Amethods%20%3D%20index.entry%28%22foo%22%2C%20Index%3A%3AMethod%29%0AT.reveal_type%28methods%29%0A%0Avars%20%3D%20index.entry%28%22foo%22%2C%20Index%3A%3AInstanceVariable%29%0AT.reveal_type%28vars%29%0A%0Aindex.entry%28%22foo%22%2C%20String%29).